src/openAPI.ts

```diff
@@ -0,0 +1,47 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as yaml from 'js-yaml';
+
+export interface OpenAPIDocument {
+    openapi: string;
+    info: object;
+    paths: object;
+    components?: {
+        schemas?: Record<string, object>;
+        [key: string]: unknown;
+    };
+    [key: string]: unknown;
+}
+
+/**
+ * Downloads the OpenAPI document YAML and returns the parsed object
+ */
+export async function getOpenAPIDocumentYAML(): Promise<OpenAPIDocument> {
+    const url = 'https://raw.githubusercontent.com/modelcontextprotocol/registry/refs/tags/v1.2.3/docs/reference/api/openapi.yaml';
```

Comment:
issue(blocking): please move this runtime fetch to the generation, like 

https://github.com/kortex-hub/mcp-registry-types/blob/754cf27da9b1e667962db1dc9f793abd2a20fb77/package.json#L16

Reply:
Thank you